src/workerd/api/sql.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/jsg/jsg.h>
+#include "basics.h"
+
+namespace workerd::api {
+
+typedef kj::OneOf<kj::Array<const byte>, kj::String, double> ValueBind;
+class SqlDatabase;
+
+class SqlResult final: public jsg::Object {
+public:
+  SqlResult(SqliteDatabase::Query& query);
+
+  kj::ArrayPtr<jsg::Dict<SqliteDatabase::Query::ValueOwned>> getResults(jsg::Lock& js);
+
+  JSG_RESOURCE_TYPE(SqlResult, CompatibilityFlags::Reader flags) {
+    JSG_READONLY_PROTOTYPE_PROPERTY(results, getResults);
+  }
+
+private:
+  kj::Vector<jsg::Dict<SqliteDatabase::Query::ValueOwned>> results;
+};
+
+class SqlPreparedStatement final: public jsg::Object {
```

Comment:
From an API perspective, it would be interesting to have `SqlDatabase::prepare` instead return a JavaScript function that, when called executes the prepared statement rather than having `SqlPreparedStatement` itself be a separate kind of object... e.g. something like the following in JavaScript:

```js
const statement = database.prepare(query)
statement(valueBind1, valueBind2, valueBind3, ...);
```